infra/modules/azure_app_service_plan_autoscaler/variables.tf

```diff
@@ -16,13 +16,54 @@ variable "autoscale_name" {
 
 variable "target_service" {
   type = object({
-    app_service_name  = optional(string)
-    function_app_name = optional(string)
+    app_service_id       = optional(string)
+    app_service_name     = optional(string)
+    function_app_id      = optional(string)
+    function_app_name    = optional(string)
+    app_service_plan_id  = optional(string)
+    location = optional(string)
   })
 
+  description = <<EOT
+  Configuration for the target service (App Service or Function App) to which the autoscaler will be applied.
+  
+  For existing services:
+  - Provide either 'app_service_name' or 'function_app_name'.
+  
+  For new services being created concurrently:
+  - Provide 'app_service_id' or 'function_app_id' (depending on the service type).
+  - 'app_service_plan_id' and 'location' must also be provided.
+
+  Note: Only one service type (App Service or Function App) should be specified.
+  EOT
+
   validation {
```

Comment:
see comment above to simplify this logic. I think this would be a breaking change anyway.